pkg/plugins/gateway/gateway_rsp_headers.go

```diff
@@ -40,12 +40,20 @@ func (s *Server) HandleResponseHeaders(ctx context.Context, requestID string, mo
 		}
 	}()
 
-	headers := []*configPb.HeaderValueOption{{
-		Header: &configPb.HeaderValue{
-			Key:      HeaderWentIntoReqHeaders,
-			RawValue: []byte("true"),
+	headers := []*configPb.HeaderValueOption{
+		{
+			Header: &configPb.HeaderValue{
+				Key:      HeaderWentIntoReqHeaders,
+				RawValue: []byte("true"),
+			},
+		},
+		{
```

Comment:
this is a separate issue, adding request to response, right? not related to status check but just putting into the same PR

Reply:
Yes. This change is to ensure request id is returned for both scenarios of routing, 1) default httproute based and 2) routing strategy is set. I identified request id as missing when testing for this PR.